packages/yoroi-extension/chrome/extension/background/handlers/content/rpc.js

```diff
@@ -0,0 +1,731 @@
+import LocalStorageApi, {
+  loadSubmittedTransactions,
+} from '../../../../../app/api/localStorage';
+import { RustModule } from '../../../../../app/api/ada/lib/cardanoCrypto/rustLoader';
+import { Logger } from '../../../../../app/utils/logging';
+import { NotEnoughMoneyToSendError, } from '../../../../../app/api/common/errors';
+import {
+  type PendingSignData,
+  type AccountBalance,
+  type Value,
+  type Address,
+  APIErrorCodes,
+  asPaginate,
+  asTokenId,
+  asValue,
+  ConnectorError,
+  DataSignErrorCodes,
+} from '../../../connector/types';
+import type {
+  IFetcher as CardanoIFetcher
+} from '../../../../../app/api/ada/lib/state-fetch/IFetcher.types';
+import { getCardanoStateFetcher, } from '../../utils';
+import AdaApi, { getAddressing } from '../../../../../app/api/ada';
+import { find721metadata } from '../../../../../app/utils/nftMetadata';
+import {
+  connectorCreateCardanoTx,
+  connectorGetBalance,
+  connectorGetCardanoRewardAddresses,
+  connectorGetChangeAddress,
+  connectorGetCollateralUtxos,
+  connectorGetUnusedAddresses,
+  connectorGetUsedAddressesWithPaginate,
+  connectorGetUtxosCardano,
+  connectorRecordSubmittedCardanoTransaction,
+  connectorSendTxCardano,
+  connectorGetAssets,
+  getTokenMetadataFromIds,
+  MAX_COLLATERAL,
+  connectorGetDRepKey,
+  connectorGetStakeKey,
+  transformCardanoUtxos,
+  assetToRustMultiasset,
+} from '../../../connector/api';
+import { authSignHexPayload } from '../../../../../app/connector/api';
+import {
+  getCardanoHaskellBaseConfig,
+} from '../../../../../app/api/ada/lib/storage/database/prepackaged/networks';
+import {
+  sendToInjector,
+  getBoundsForTabWindow,
+  popupProps,
+} from './utils';
+import { asGetAllUtxos, } from '../../../../../app/api/ada/lib/storage/models/PublicDeriver/traits';
+import { asAddressedUtxo as asAddressedUtxoCardano, } from '../../../../../app/api/ada/transactions/utils';
+import {
+  getConnectedSite,
+  setConnectedSite,
+  findWhitelistedConnection,
+  connectError,
+  getConnectedWallet,
+  type ConnectedSite,
+  type SignContinuationDataType,
+} from './connect';
+import type { CardanoTxRequest } from '../../../../../app/api/ada';
+import type { NFTMetadata } from '../../../../../app/api/ada/lib/storage/database/primitives/tables';
+
+declare var chrome;
+
+async function confirmSign(
+  tabId: number,
+  request: PendingSignData,
+  connectedSite: ConnectedSite,
+  continuationData: SignContinuationDataType,
+  uid: number,
+): Promise<void> {
+  const bounds = await getBoundsForTabWindow(tabId);
+
+  connectedSite.pendingSigns[String(request.uid)] = {
+    request,
+    openedWindow: false,
+    continuationData,
+    uid,
+  };
+  await setConnectedSite(tabId, connectedSite);
+  chrome.windows.create({
+    ...popupProps,
+    url: chrome.runtime.getURL(`/main_window_connector.html#/signin-transaction`),
+    left: (bounds.width + bounds.positionX) - popupProps.width,
+    top: bounds.positionY + 80,
+  });
```

Comment:
I think `return` or `await` is missing. The function itself is async and returns a promise, but this last async call result is ignored so the function might `.then` before `windows.create` actually finished

Reply:
This is OK because from the RPC handler function's perspective, popping up the window is the last action. The whole process will resume when the popup window sends further messages.

Actually, under mv3, the service worker may stop after this.